### PR TITLE
[DoctrineBridge] Bumps doctrine/persistence to version 2.x

### PR DESCRIPTION
**Pull request by fruitwasp**
| Q             | A
| ------------- | ---
| Branch?       | master (bumps to a new major version)
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

Adds support for doctrine/persistence version to 2.x. As it's a new major version, I've targeted the master branch. Relates to https://github.com/doctrine/DoctrineBundle/pull/1181

Awating pull request to be merged in dependency doctrine/data-fixtures: https://github.com/doctrine/data-fixtures/pull/349

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
         "php": ">=7.2.5",
         "ext-xml": "*",
         "doctrine/event-manager": "~1.0",
-        "doctrine/persistence": "^1.3",
+        "doctrine/persistence": "^2.0",
```
Comment (derrabus) [inline composer.json:22]: Why not `^1.3|^2.0`?

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by nicolas-grekas** [comment]
This should target branch 3.4 and compat with 1.* should be preserved.

**Comment by nicolas-grekas** [comment]
Closing as staled, please resubmit when ready.

**Comment by derrabus** [comment]
I've had a quick look.

> This should target branch 3.4

On the 3.4 branch, we don't have any version constraints for doctrine/persistence.

On 4.4 and above, we're still blocked by `doctrine/data-fixtures`.

https://github.com/doctrine/data-fixtures/blob/ba588dfae2da361e8cb4908d2b2a21d09a47e8af/composer.json#L19

**Comment by derrabus** [comment]
Continuation: #37639

**Pull request closed without merging**